### PR TITLE
Update Judgee Tracking E2E Test

### PR DESCRIPTION
**Pull request by yifan1207**
- Changed test file to pytest
- Added organization ID to user headers to resolve 422 errors
- Improved Test Assertions: Modified count verification to check for increments rather than exact values
- Created simplified test case that avoids 500 errors
- Updated User/Org Tests: Adjusted expectations to reflect shared organization tracking

UPDATE (3/4 4:04 PM):

- Created and updated E2E tests to tests traces tracking and added edge case testing for judgee tracking

### CODE REVIEW

File: src/e2etests/test_judgee_update.py [hunk 1]
```diff
@@ -1,598 +1,756 @@
 """
```
Comment (JCamyre) [inline src/e2etests/test_judgee_update.py:1]: May need to update tests depending on the implementation changes you make on `judgment` side.
